numba/cuda/stubs.py

```diff
@@ -584,6 +594,156 @@ class habs(Stub):
 
         """
 
+    class hsin(Stub):
+        """hsin(a)
+
+        Calculate sine in round to nearest even mode. Supported on fp16
+        operands only.
+
+        Returns the sine result.
+
+        """
+
+    class hcos(Stub):
+        """hsin(a)
+
+        Calculate cosine in round to nearest even mode. Supported on fp16
+        operands only.
+
+        Returns the cosine result.
+
+        """
+
+    class hlog(Stub):
+        """hlog(a)
+
+        Calculate natural logarithm in round to nearest even mode. Supported
+        on fp16 operands only.
+
+        Returns the natural logarithm result.
+
+        """
+
+    class hlog10(Stub):
+        """hlog10(a)
+
+        Calculate logarithm base 10 in round to nearest even mode. Supported
+        on fp16 operands only.
+
+        Returns the logarithm base 10 result.
+
+        """
+
+    class hlog2(Stub):
+        """hlog2(a)
+
+        Calculate logarithm base 2 in round to nearest even mode. Supported
+        on fp16 operands only.
+
+        Returns the logarithm base 2 result.
+
+        """
+
+    class hexp(Stub):
+        """hexp(a)
+
+        Calculate natural exponential, (2.71828 ** a) in round to nearest mode.
```

Comment:
I think this calculates with more precision than the docstring implies. How about:

```suggestion
        Calculate natural exponential, exp(a), in round to nearest mode.
```